src/extension.ts

```diff
@@ -94,3 +94,13 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
 export function deactivateInternal(): void {
     ext.diagnosticWatcher?.dispose();
 }
+
+function setupEvents(context: vscode.ExtensionContext): void {
+    // Event emitter for when a group is focused/unfocused
```

Comment:
@alexweininger I moved these down here, I could see this list getting long and having it in the main body of the activate function would clutter it a bit.

Reply:
Also I followed the `ext.emitters` / `ext.events` pattern you established.